firmware/boron-ins-fsm/test/imDoorSensorTests.cpp

```diff
@@ -0,0 +1,33 @@
+#define CATCH_CONFIG_MAIN
+#include "base.h"
+#include "../src/imDoorSensor.cpp"
+#include "../src/imDoorSensor.h"
+#include "../src/flashAddresses.h"
+
+#define NO_OF_TEST_DOOR_STATUS 8
+
+SCENARIO("isDoorOpen", "[isDoorOpen]") {
+    GIVEN("A variety of IM door status advertising data") {
+        int doorStatusOpen[NO_OF_TEST_DOOR_STATUS] = { 0x02, 0x03, 0x06, 0x07, 0x0A, 0x0B, 0x0E, 0x0F };
+        int doorStatusClosed[NO_OF_TEST_DOOR_STATUS] = { 0x00, 0x01, 0x04, 0x05, 0x08, 0x09, 0x0C, 0x0D };
+
+        WHEN("The function is called with the door status indicating open") {
+
+            THEN("The function should return 1") {
+                for (int i = 0; i < NO_OF_TEST_DOOR_STATUS; ++i) {
+                    REQUIRE(isDoorOpen(doorStatusOpen[i]) == 1);
+                }
```

Comment:
This is a great catch. I'll look into making these changes

Reply:
I changed this so the test case number is output for each loop. I can also confirm that even if one test case fails, all the other ones are still run and reported on

Reply:
Cool. I didn't know about the `INFO` function. That totally works.

However, I would find this more useful if it output the value of `doorStatusOpen[i]` or `doorStatusClosed[i]` instead of `i`. What do you think?

Reply:
I completely agree with you and I have made the change in my latest commit.